### PR TITLE
Add CombinedEdgeLabelMapping

### PR DESCRIPTION
**Pull request by BentolhodaH**
Add CombinedEdgeLabelMapping and its tests
Encapsulate creating EdgeLabelMapping into different functions to avoid duplicate code 
Add UnSupportedEdgeLabelException

### CODE REVIEW

**Comment by BentolhodaH** [comment]
In 'RunBGPOverNeo4j.java,' currently, we are not reading the TTL file. It is necessary to specify a command-line argument to obtain the file address, similar to the file query file address. In this step, should I make modifications to that part to read from the example file instead of creating classes manually?

**Comment by hartig** [comment]
Indeed. The command-line tool needs to be extended to provide such an argument. Please do that, but in a separate PR. In terms of implementing it, it should be implemented more like the argument for the federation catalog that we have in the other command-line tool.

File: src/main/java/se/liu/ida/hefquin/engine/wrappers/lpgwrapper/impl/CombinedEdgeLabelMappingToURIsImpl.java [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+    protected final List<EdgeLabelMapping> edgeLabelMappings;
+
+    public CombinedEdgeLabelMappingToURIsImpl(final List<EdgeLabelMapping> edgeLabelMappings){
+        this.edgeLabelMappings = edgeLabelMappings;
+    }
+
+    @Override
+    public Node map(final String label) {
+        for (EdgeLabelMapping edgeLabelMapping : edgeLabelMappings) {
```
Comment (hartig) [inline src/main/java/se/liu/ida/hefquin/engine/wrappers/lpgwrapper/impl/CombinedEdgeLabelMappingToURIsImpl.java:18]:
```suggestion
        for ( final EdgeLabelMapping edgeLabelMapping : edgeLabelMappings ) {
```

File: src/main/java/se/liu/ida/hefquin/engine/wrappers/lpgwrapper/impl/CombinedEdgeLabelMappingToURIsImpl.java [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+            try {
+                return edgeLabelMapping.map(label);
+            }
+            catch (IllegalArgumentException exception) {}
+        }
+        throw new UnSupportedEdgeLabelException("The given edge label (" + label + ") is not a supported label in the image of this edge label mapping.");
+    }
+
+    public String unmap(final Node node) {
+        for (EdgeLabelMapping edgeLabelMapping : edgeLabelMappings) {
```
Comment (hartig) [inline src/main/java/se/liu/ida/hefquin/engine/wrappers/lpgwrapper/impl/CombinedEdgeLabelMappingToURIsImpl.java:28]:
```suggestion
        for ( final EdgeLabelMapping edgeLabelMapping : edgeLabelMappings ) {
```

File: src/main/java/se/liu/ida/hefquin/engine/wrappers/lpgwrapper/impl/CombinedEdgeLabelMappingToURIsImpl.java [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+                return edgeLabelMapping.unmap(node);
+            }
+            catch (UnSupportedEdgeLabelException exception) {}
+        }
+        throw new UnSupportedEdgeLabelException("The given RDF term (" + node.toString() + ") is not an URI node or not in the image of this edge label mapping.");
+    }
+
+    @Override
+    public boolean isPossibleResult(final Node node) {
+        for (EdgeLabelMapping edgeLabelMapping : edgeLabelMappings) {
```
Comment (hartig) [inline src/main/java/se/liu/ida/hefquin/engine/wrappers/lpgwrapper/impl/CombinedEdgeLabelMappingToURIsImpl.java:39]:
```suggestion
        for ( final EdgeLabelMapping edgeLabelMapping : edgeLabelMappings ) {
```

File: src/main/java/se/liu/ida/hefquin/engine/wrappers/lpgwrapper/impl/CombinedEdgeLabelMappingToURIsImpl.java [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+        this.edgeLabelMappings = edgeLabelMappings;
+    }
+
+    @Override
+    public Node map(final String label) {
+        for (EdgeLabelMapping edgeLabelMapping : edgeLabelMappings) {
+            try {
+                return edgeLabelMapping.map(label);
+            }
+            catch (IllegalArgumentException exception) {}
```
Comment (hartig) [inline src/main/java/se/liu/ida/hefquin/engine/wrappers/lpgwrapper/impl/CombinedEdgeLabelMappingToURIsImpl.java:22]: Shouldn't this be `UnSupportedEdgeLabelException` ?

**Review by hartig — CHANGES_REQUESTED** [review]
A few minor things.

**Review by hartig — APPROVED** [review]
Thanks!!

**Pull request merged**